platform/drm/cog-drm-gles-renderer.c

```diff
@@ -0,0 +1,567 @@
+/*
+ * cog-drm-gles-renderer.c
+ * Copyright (C) 2021 Igalia S.L.
+ *
+ * Distributed under terms of the MIT license.
+ */
+
+#include "../../core/cog.h"
+#include "../common/cog-gl-utils.h"
+#include "cog-drm-renderer.h"
+#include <drm_fourcc.h>
+#include <drm_mode.h>
+#include <epoxy/gl.h>
+#include <errno.h>
+#include <gbm.h>
+#include <glib-unix.h>
+#include <wayland-util.h>
+#include <wpe/fdo-egl.h>
+#include <wpe/fdo.h>
+#include <xf86drm.h>
+#include <xf86drmMode.h>
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(drmModePlane, drmModeFreePlane)
+
+typedef struct {
+    CogDrmRenderer base;
+
+    struct gbm_device  *gbm_device;
+    struct gbm_surface *gbm_surface;
+    struct gbm_bo      *current_bo;
+    struct gbm_bo      *next_bo;
+    uint32_t            gbm_format;
+
+    /*
+     * Logical view size without transformations applied, which is needed to
+     * change the transformed size (i.e. rotated) of the view after the view
+     * backend has been instantiated. Note that physical output size is
+     * always determinted from the chosen DRM/KMS mode.
+     */
+    uint32_t width, height;
+
+    CogDrmRendererRotation rotation;
+
+    EGLDisplay egl_display;
+    EGLConfig  egl_config;
+    EGLContext egl_context;
+    EGLSurface egl_surface;
+
+    GLuint gl_program;
+    GLuint gl_texture;
+    GLint  gl_attrib_position;
+    GLint  gl_attrib_texture;
+    GLint  gl_uniform_texture;
```

Comment:
I am writing the following next to the variables for my updated patch set:

```c
    /*
     * The renderer wraps frames provided by wpebackend-fdo into a texture,
     * then paints a quad which samples said texture. A simple GLSL shader
     * program uses the "position" attribute to fetch the coordinates of
     * the quad, and the "texture" attribute the UV mapping coordinates for
     * the texture. Rotation is achieved by changing the UV mapping. The
     * "texture" uniform is used to reference the texture unit where the
     * frame textures get loaded.
     */
```

Please let me know if this may need further clarification.